packages/gatsby/src/redux/reducers/schema-customization.js

```diff
@@ -21,9 +21,20 @@ module.exports = (
     case `CREATE_TYPES`: {
       let types
       if (_.isArray(action.payload)) {
-        types = [...state.types, ...action.payload]
+        types = [
+          ...state.types,
+          ...action.payload.map(typeOrTypeDef => {
+            return {
+              typeOrTypeDef,
+              plugin: action.plugin,
```

Comment:
What is the shape of `plugin` here? If it is more than `plugin.name`, we can just pluck off the name.

Reply:
Good idea.